server/coordinator/scheduler/manager/scheduler_manager.go

```diff
@@ -260,13 +279,51 @@ func (m *ManagerImpl) UpdateDeployMode(ctx context.Context, enable bool) error {
 	return nil
 }
 
-func (m *ManagerImpl) GetDeployMode(_ context.Context) (bool, error) {
+func (m *schedulerManagerImpl) GetDeployMode(_ context.Context) (bool, error) {
 	m.lock.RLock()
 	defer m.lock.RUnlock()
 
 	if m.topologyType != storage.TopologyTypeDynamic {
-		return false, errors.WithMessage(ErrInvalidTopologyType, "deploy mode could only get when topology type is dynamic")
+		return false, ErrInvalidTopologyType.WithCausef("deploy mode could only get when topology type is dynamic")
 	}
 
 	return m.deployMode, nil
 }
+
+func (m *schedulerManagerImpl) AddShardAffinityRule(ctx context.Context, rule scheduler.ShardAffinityRule) error {
+	var firstErr error
+	for _, scheduler := range m.registerSchedulers {
+		if err := scheduler.AddShardAffinityRule(ctx, rule); err != nil {
+			firstErr = err
```

Comment:
```suggestion
			firstErr = err
```
Add some log get error msg.

Reply:
I guess there is no need to print the error on this level?

Reply:
Yes, waring or info is enough, what I mean is that just adding a line of log here.